src/platform/Linux/ConfigurationManagerImpl.cpp

```diff
@@ -140,7 +140,8 @@ CHIP_ERROR ConfigurationManagerImpl::GetPrimaryWiFiMACAddress(uint8_t * buf)
     VerifyOrExit(getifaddrs(&addresses) == 0, error = CHIP_ERROR_INTERNAL);
     for (auto addr = addresses; addr != nullptr; addr = addr->ifa_next)
     {
-        if ((addr->ifa_addr) && (addr->ifa_addr->sa_family == AF_PACKET) && strncmp(addr->ifa_name, "lo", IFNAMSIZ) != 0)
+        if ((addr->ifa_addr) && (addr->ifa_addr->sa_family == AF_PACKET) &&
+            strncmp(addr->ifa_name, CHIP_DEVICE_CONFIG_WIFI_STATION_IF_NAME, IFNAMSIZ) == 0)
```

Comment:
Won't it cause many failures when running Linux apps such as `chip-lighting-app` in Ubuntu that uses `wlo1` for the Wi-Fi interface name? `CHIP_DEVICE_CONFIG_WIFI_STATION_IF_NAME` defaults to `wlan0`.

Reply:
Hmm, I think you're right. I can update to prioritize interface name match and fallback to use the first non-loopback interface (current behavior) if no interface match.

So for the devices that do specify the proper/right CHIP_DEVICE_CONFIG_WIFI_STATION_IF_NAME, it will guarantee the proper MAC address will be used.

Let me know what you think.

Thanks,
Peter,